data/transactions/logic/ledger_test.go

```diff
@@ -826,9 +826,11 @@ func (l *Ledger) appl(from basics.Address, appl transactions.ApplicationCallTxnF
 	}
 	pass, cx, err := EvalContract(params.ApprovalProgram, gi, aid, ep)
 	if err != nil {
+		ad.EvalDelta = transactions.EvalDelta{}
 		return err
 	}
 	if !pass {
+		ad.EvalDelta = transactions.EvalDelta{}
 		return errors.New("Approval program failed")
 	}
 	ad.EvalDelta = cx.txn.EvalDelta
```

Comment:
The real ledger does not set EvalDeltas if an error happens, however the test ledger implicitly does right now. This is because `ad` is pointer to `cx.txn.ApplyData`, so this line is a noop.

I added some lines above this to explicitly clear the AD's EvalDelta if a failure happens, and that's good enough to get my tests to pass.

Reply:
maybe add this comment inline